src/detect-ssh-proto-version.c

```diff
@@ -318,27 +314,19 @@ static int DetectSshVersionTestParse03 (void)
 {
     DetectSshVersionData *ssh = NULL;
     ssh = DetectSshVersionParse(NULL, "2_com");
-    if (ssh != NULL) {
-        DetectSshVersionFree(NULL, ssh);
-        return 0;
-    }
+    FAIL_IF_NOT_NULL(ssh);
     ssh = DetectSshVersionParse(NULL, "");
-    if (ssh != NULL) {
-        DetectSshVersionFree(NULL, ssh);
-        return 0;
-    }
+    FAIL_IF_NOT_NULL(ssh);
     ssh = DetectSshVersionParse(NULL, ".1");
-    if (ssh != NULL) {
-        DetectSshVersionFree(NULL, ssh);
-        return 0;
-    }
+    FAIL_IF_NOT_NULL(ssh);
     ssh = DetectSshVersionParse(NULL, "lalala");
+    FAIL_IF_NOT_NULL(ssh);
+
     if (ssh != NULL) {
         DetectSshVersionFree(NULL, ssh);
-        return 0;
     }
```

Comment:
Make sure to remove all the calls that are covered by the FAIL/PASS API.

Reply:
How this call is covered by the FAIL/PASS API?

Reply:
I suppose you are asking about the `DetectSshVersionFree()` call.

With the new FAIL/PASS approach, engine resources must only be freed when a test succeeds. But, since this test if for ensuring the `ssh` _is_ null, if the test succeeds, there's nothing to be freed. And if the test fails, well, we won't call freeing statements either.

If your question was about the whole if statement: all if statements should be converted to the new fail/pass API ;)

Reply:
It was about DetectSshVersionFree() call. I understand now, Thank you!